app/controllers/images_controller.rb

```
@@ -30,4 +30,14 @@ def index
       @images = Image.all.order('created_at Desc')
     end
   end
+
+  def destroy
+    @image = Image.find(params[:id])
+    @image.destroy
+    flash[:success] = 'Delete success'
+    redirect_to images_path
+  rescue ActiveRecord::RecordNotFound
+    flash[:success] = 'Delete success'
+    redirect_to images_path
```

we could reduce some duplication here by calling these 2 lines after the `rescue`. We would have to use a `begin`/`end` block to do so:
```
begin
  @image = Image.find(params[:id])
  @image.destroy
rescue ActiveRecord::RecordNotFound
end
flash[:success] = 'Delete success'
. . .
```
Since it's outside of the block, it will execute whether there is an exception or not.
Alternatively, we could use `find_by` instead of `find`, which won't raise an exception, and then check to make sure `@image` exists:
```
@image = Image.find_by(id: params[:id])
@image&.destroy
flash[:success] = 'Delete success'
...
```
The `&` before the `.destroy` is called a safe navigation operator; it ensures that if the object the method is being called on is not nil before calling the method, otherwise it returns nil. While `nil.destroy` would raise an exception, `nil&.destroy` will silently return `nil` without breaking.

---

Wow, thats very useful to learn. Thanks for sharing!